github/resource_github_actions_environment_secret.go

```diff
@@ -0,0 +1,192 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"log"
+	"net/http"
+
+	"github.com/google/go-github/v35/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceGithubActionsEnvironmentSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubActionsEnvironmentSecretCreateOrUpdate,
+		Read:   resourceGithubActionsEnvironmentSecretRead,
+		Delete: resourceGithubActionsEnvironmentSecretDelete,
+
+		Schema: map[string]*schema.Schema{
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"environment": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"secret_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateSecretNameFunc,
+			},
+			"encrypted_value": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
```

Comment:
See also https://github.com/integrations/terraform-provider-github/pull/807/files#r645477509 as it applies here.